### PR TITLE
[User model] JWT token handling

### PR DESCRIPTION
**Pull request by emawby**
# Description
## One Line Summary
Using the JWT token for a user and adding the onJwtExpired method to User.

## Details
- Adding onJwtExpiredHandler to the User namespace
- Adding the JWT token header to user requests that require it in the prepareForExecution method.
- **Note that firing the expired handler is not currently setup since we are not handling errors yet.**

### Motivation
Enable JWT authorization for users

### Scope
User module

# Testing
## Unit testing
N/A

## Manual testing
Running on dev app

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [x] REST API requests
   - [x] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/1187)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSIdentityModel.swift [hunk 1]
```diff
@@ -39,6 +39,9 @@ class OSIdentityModel: OSModel {
     }
 
     var aliases: [String: String] = [:]
+    
+    // TODO: We need to make this token secure
+    public var jwtBearerToken: String?
```
Comment (fhboswell) [inline iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSIdentityModel.swift:44]: [Keychain](https://developer.apple.com/documentation/security/keychain_services)?

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift [hunk 2]
```diff
@@ -807,3 +815,14 @@ class OSRequestDeleteSubscription: OneSignalRequest, OSUserRequest {
         _ = prepareForExecution()
     }
 }
+
+internal extension OneSignalRequest {
+    func addJWTHeader(identityModel: OSIdentityModel) {
+        guard let token = identityModel.jwtBearerToken else {
+            return
```
Comment (fhboswell) [inline iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift:822]: May be good to have some message logged here

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift [hunk 3]
```diff
@@ -386,6 +392,19 @@ public class OneSignalUserManagerImpl: NSObject, OneSignalUserManager {
             identityModel: identityModel
         )
     }
+    
+    
+    private func fireJwtExpired() {
```
Comment (fhboswell) [inline iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift:397]: JWT jwt and Jwt is used I suggest removing Jwt and sticking with the other two

**Review by fhboswell — CHANGES_REQUESTED** [review]

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift (hunk 3 above)
Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift:397]: @fhboswell there is jwt and Jwt being used as we are using camelcase. We could use JWT, but probably a bit easier to read as camelcase.

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSIdentityModel.swift (hunk 1 above)
Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSIdentityModel.swift:44]: Keychain normally isn't used for JWT. JWTs are used designed to be short lived (normally 1 hour), so I don't think we need to do anything extra to secure it.

**Review by jkasten2 — APPROVED** [review]

File: iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift (hunk 3 above)
Comment (fhboswell) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift:397]: @jkasten2 
Jwt if fine if it is preferred however I think we should be consistent. We use JWT in OSUserRequest as part of this PR.

` func addJWTHeader(identityModel: OSIdentityModel) {`

Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/OneSignalUser/Source/OneSignalUserManagerImpl.swift:397]: Ah I see, yes we should address that in another PR @emawby @nan-li

**Pull request merged**